src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -142,6 +154,10 @@ internal partial class InProcLanguageServer : IAsyncDisposable
                 logger);
         }
 
+        // Make sure this isn't inlined so these types are only loaded
+        // after the type check in CreateAsync.
+        // Removal tracked by https://github.com/dotnet/roslyn/issues/52454
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
And I'll say: the CLR has a bunch of rules about what it does and doesn't inline, so I presume this _won't_ inline no matter what. But better to be explicit, and the NoInlining pattern is common enough around dealing with potentially missing types that it makes it a bit more clear what we're doing.